examples/model_selection/plot_nested_cross_validation_iris.py

```diff
@@ -0,0 +1,119 @@
+"""
+=========================================
+Nested versus Non-Nested cross-validation
+=========================================
+
+This example compares non-nested and nested cross-validation strategies on a
+classifier of the iris data set. Nested cross-validation (CV) is often used to
+train a model in which hyperparameters also need to be optimized. Nested CV
+approximates the generalization error of the resulting estimator of the
```

Comment:
We're inconsistent on this terminology. Hyperparameter is a Bayesian term that's recently become more popular, for clarity, in the rest of the ML community. But scikit-learn uses "get_params" and "set_params" to operate on these things and not on the model parameters in the bayesian sense, so I find it a bit strange to call it a hyperparameter.